connectors/src/connectors/google_drive/index.ts

```diff
@@ -684,6 +672,25 @@ export async function retrieveGoogleDriveContentNodes(
     { concurrency: 4 }
   );
 
+  const drivesOrTopLevelFolderNodes = await (async () => {
+    if (drivesOrTopLevelFolders.length === 0) {
+      return [];
+    }
+    const c = await ConnectorResource.fetchById(connectorId);
+    if (!c) {
+      logger.error({ connectorId }, "Connector not found");
+      throw new Error("Connector not found");
+    }
+    const authCredentials = await getAuthObject(c.connectionId);
+    return removeNulls(
+      await getFoldersAsContentNodes({
```

Comment:
Is there any chance we can avoid calling Google Drive API to render the tree view?

Reply:
IMHO, not easily (or maybe really hacky, ugly stuff)

Long term we could merge the google_drive_folders table into the google_drive_files one (which contains folders too) but that would be much more involved

Intermediate: store more data in the google_drive_folders table so we can get everything from there, but will take a bit of time too 

To note: we already call it ~often, e.g. when editing google drive source of an assistant, or any call to connectorsPermission on a google drive datasource

Reply:
Seen IRL, not ideal but already existing. Follow up item to open an issue and improve the situation.